x-pack/plugins/security/server/session_management/session.ts

```diff
@@ -194,6 +197,26 @@ export class Session {
       return { error: new SessionUnexpectedError(), value: null };
     }
 
+    // The only reason why we check if the session is within the concurrent session limit _after_ decryption
+    // is to record decrypted username and profile id in the audit logs.
+    const isSessionWithinConcurrentSessionLimit =
```

Comment:
__note:__ Initially, the code was enforcing the limit at creation time, but after various load tests I noticed that it was easy to bypass the limit if user creates multiple sessions at the same time leading to the race conditions in the limit enforcing code. If instead, we enforce the limit at the "fetch" time we'd never allow using a session that's outside the limit.